win32ss/user/user32/misc/exticon.c

```diff
@@ -323,6 +323,42 @@ static UINT ICO_ExtractIconExW(
 	}
 	CloseHandle(fmapping);
 
+#ifdef __REACTOS__
+    /* Check if the resource is an animated icon/cursor */
+    if (!memcmp(peimage, "RIFF", 4))
```

Comment:
```suggestion
    if (fsizel >= ... && !memcmp(peimage, "RIFF", 4))
```

Reply:
Boundary check added a few lines below this. Only one boundary check should be enough. Thanks.

Reply:
If `fsizel` was zero, what will happen?

Reply:
I checked these cases with the code as it was without the fsizel test and for a zero byte file there was a 'CreateFileMapping error 1006' before we got to any of the new code. I checked with sizes of 1, 2, 3, and 4 and there were no crashes or exceptions, so obviously these errors were most likely caught in another part of the code and I was too lazy to track down. I did add a general requirement that fsizel >= 52 which should help.

Reply:
Ok, then fix the typos of comments.